src/redux/missions/missionSlice.js

```diff
@@ -0,0 +1,72 @@
+import { createSlice, createAsyncThunk, isRejectedWithValue } from '@reduxjs/toolkit';
+
+const url = 'https://api.spacexdata.com/v3/missions';
+
+const initialState = {
+  missions: [],
+  isLoading: false,
+  error: null,
+};
+
+export const getMissions = createAsyncThunk('missions/getMissions', async () => {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error('Network response was not OK');
+    }
+    const data = await response.json();
+    const missionArr = Object.keys(data).map((key) => ({
+      mission_id: key,
+      reserved: false,
+      ...data[key],
+    }));
+    return missionArr;
+  } catch (error) {
+    return isRejectedWithValue(error);
+  }
+});
```

Comment:
- Hi, I notice that When I refresh any of the pages (Rocket, Mission, or My profile), all the selections I have done or have selected on different pages disappear, including the view on my profile page. I would advise you to go through your code to fix that, then submit it again for a re-review.
![Capture](https://github.com/joyapisi/space-traveler-hub/assets/63972670/c7ae8e89-234f-4d1d-9f3f-fbdaf03c7bcf)

Reply:
@OLIPLICHE This was not a scope requirement of the project. It clearly states that we are not preserving user data in the **technical setup** section. So I believe this is an unfair review

Reply:
Here is a screenshot of the technical requirements.

<img width="815" alt="Screenshot 2023-05-17 at 16 16 45" src="https://github.com/joyapisi/space-traveler-hub/assets/69506217/b4195440-9c28-4a2d-85ca-71948fb0a4f8">